server/src/main/java/io/crate/planner/optimizer/rule/RewriteNestedLoopJoinWithConstantConditions.java

```diff
@@ -0,0 +1,113 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.planner.optimizer.rule;
+
+import static io.crate.planner.optimizer.matcher.Pattern.typeOf;
+import static io.crate.planner.optimizer.rule.FilterOnJoinsUtil.getNewSource;
+
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Set;
+
+import io.crate.analyze.relations.QuerySplitter;
+import io.crate.expression.operator.AndOperator;
+import io.crate.expression.symbol.Symbol;
+import io.crate.metadata.NodeContext;
+import io.crate.metadata.RelationName;
+import io.crate.metadata.TransactionContext;
+import io.crate.planner.consumer.RelationNameCollector;
+import io.crate.planner.node.dql.join.JoinType;
+import io.crate.planner.operators.HashJoin;
+import io.crate.planner.operators.LogicalPlan;
+import io.crate.planner.operators.NestedLoopJoin;
+import io.crate.planner.optimizer.Rule;
+import io.crate.planner.optimizer.matcher.Captures;
+import io.crate.planner.optimizer.matcher.Pattern;
+import io.crate.statistics.TableStats;
+
+public class RewriteNestedLoopJoinWithConstantConditions implements Rule<NestedLoopJoin> {
```

Comment:
```suggestion
public class MoveConstantJoinConditionsBeneathNestedLoop implements Rule<NestedLoopJoin> {
```

Would fit better into the existing naming scheme of the rules